Small clean up of troubleshooting page (#2399)
nf-core/website: src/content/docs/usage/troubleshooting.md

subtitle: How to troubleshoot common mistakes and issues
---

## How to read this page

> 1. Search your specific error message in this page using your browsers search functionality (e.g. <kbd>ctrl</kbd> + <kbd>f</kbd>)

Reviewer: ```suggestion
1. Search your specific error message in this page using your browsers search functionality (e.g. <kbd>ctrl</kbd> + <kbd>f</kbd> or <kbd>cmd</kbd> + <kbd>f</kbd>)
```

Note that if your sample name is "messy" then you have to be very particular with your glob specification (see point 2 above). A file name like `L1-1-D-2h_S1_L002_R1_001.fastq.gz` can be difficult enough for a human to read. Specifying `*{1,2}*.gz` will not give you what you want, whilst `*{R1,R2}*.gz` will.

> Please also note pipelines with 'direct input' can't take a list of multiple input files - it takes a glob expression. If your input files are scattered in different paths then we recommend that you generate a directory with symlinked files. If running in paired end mode please make sure that your files are sensibly named so that they can be properly paired. See the previous point.

Author: ```suggestion
Please also note that genomics pipelines with 'direct input' can't take a list of multiple input files - it takes a glob expression. If your input files are scattered in different paths then we recommend that you generate a directory with symlinked files. Furthermore, you have paired-end sequencing data mode please make sure that your files are sensibly named so that they can be properly paired e.g. (--input '*_{R1,R2}.fastq.gz').
```

> 

Reviewer: the part about paried end is too sequencing specific imo and difficult to understand for non genomics people

Author: @mashehu what about the suggestion abvove?

Reviewer: yep, looks good!

```

#### Sample sheet input

If you are using a sample sheet or TSV input method, check there is not a mistake or typo in the path in a given column. Common mistakes are a trailing space at the end of the path, which can cause problems.

#### Using a local version of iGenomes

The iGenomes config file uses `params.igenomes_base` to make it possible to use a local copy of iGenomes. However, as custom config files are loaded after `nextflow.config` and the `igenomes.config` has already been imported and parsed, setting `params.igenomes_base` in a custom config file has no effect and the pipeline will use the s3 locations by default. To overcome this you can specify a local iGenomes path by either:

- Specifying an `--igenomes_base` path in your execution command.

```bash
nextflow run nf-core/<pipeline> --input <input> -c <config> -profile <profile> --igenomes_base <path>/<to>/<data>/igenomes
```

- Specifying the `igenomes_base` parameter in a parameters file provided with `-params-file` in `yaml` or `json` format.

```bash
nextflow run nf-core/<pipeline> -profile <profile> -params-file params.yml
```

Where the `params.yml` file contains the pipeline params:

> ```bash

Reviewer: ```suggestion
```yaml title="params.yml"
```

If you are still unable to resolve the issue, please make a GitHub issue on the corresponding pipeline repository.

### I get a exceeded job memory limit error

If you hit an error such as `Process requirement exceeds available memory -- req: 100 GB; avail: 15.5 GB`, this implies the pipeline run has requested more memory than is available on your system.

While Nextflow tries to make your life easier by automatically retrying jobs that run out of memory with more resources (until a [specified max-limit](https://nf-co.re/docs/usage/configuration#tuning-workflow-resources)), sometimes you may have such large data that you run out even after the default 3 retries.

To address this you will need to make a configuration file that tells Nextflow how much memory or CPUs are available on your system, and also how much memory or CPUs the failing step should use.

Please see the two following sections of the [configuring nf-core pipelines]
(https://nf-co.re/docs/usage/configuration) page:

> - Setting limits: '[Max Resources](https://nf-co.re/docs/usage/configuration#max-resources)'
> - Adusting the process resources: '[Tuning workflow resources](https://nf-co.re/docs/usage/configuration#tuning-workflow-resources)'

Reviewer: ```suggestion
- Setting limits: [Max Resources](https://nf-co.re/docs/usage/configuration#max-resources)
- Adusting the process resources: [Tuning workflow resources](https://nf-co.re/docs/usage/configuration#tuning-workflow-resources)
```
